### DISCUSSION TITLE
[Feature Request] option to hide the metrics shown in chart by default

### DISCUSSION THREAD

**Discussion by ketulgupta1995** [apache/superset#23508] [Ideas]
In all  the charts all the metrics are visible while plotting the chart .  having an option to hide a few metrics would be useful as

> This would help in reducing rework while creating charts

**Comment by SrslyJosh** [comment]
@ketulgupta1995 I'm having the same issue: I've got some datasets generated by another team, and they have metrics stored in columns with a schema like:

> timestamp, host, cpu_min, cpu_avg, cpu_max, mem_min, mem_avg, mem_max

I'd like to set up a dashboard that will allow people to toggle which metrics they're viewing, so that I don't have to create a different chart for each analysis use case.

As a work-around, I think I am going to create a new dataset with the following schema:

> timestamp, host, kpi, stat, value

Where `kpi` will be "cpu", "mem", etc. and `stat` will be "min", "avg", "max", etc.

This schema should allow me to create dashboard where users can filter to display specific KPIs and stats.

However, it would be simpler if users could just toggle which metrics are visible in a dashboard.
